substrate-metadata/src/util.ts

```diff
@@ -210,3 +211,41 @@ export function isPreV14(metadata: Metadata): boolean {
             return false
     }
 }
+
+interface PolkadotJSTypesBundle {
+    spec: {
+        [specName: string]: {
+            types: {
+                types: Record<string, OldTypeDefinition>
+                minmax: [
+                    min: number | undefined | null,
+                    max: number | undefined | null
+                ]
+            }[]
+            alias: OldTypesAlias,
+            signedExtensions: Record<string, OldTypeExp>
+        }
+    }
+}
+
+export function convertPolkadotJSTypesBundle(typesByndle: PolkadotJSTypesBundle): Record<string, OldTypesBundle> {
+    let res: Record<string, OldTypesBundle> = {}
+
+    for (let specName in typesByndle.spec) {
+        let spec = typesByndle.spec[specName]
+        res[specName] = {
+            types: {},
+            typesAlias: spec.alias,
+            signedExtensions: spec.signedExtensions,
+            versions: spec.types.map(v => ({
+                minmax: [
+                    v.minmax[0] || null,
+                    v.minmax[1] || null
+                ],
+                types: v.types,
```

Comment:
Note, that we don't support the following polkadotjs features:

* [fallbacks](https://github.com/polkadot-js/api/blob/b24e13f478418ea7447b7c02eb3bc22b7e6a5452/packages/types/src/types/definitions.ts#L13)
* [aliases](https://github.com/polkadot-js/api/blob/b24e13f478418ea7447b7c02eb3bc22b7e6a5452/packages/types/src/types/definitions.ts#L13)

We need to throw an error when such features are used.

Reply:
Maybe just warn?

Reply:
Ok. Yes, both for fallback and aliases it is enough to warn.